src/linux/processors.c

```diff
@@ -215,6 +218,26 @@ bool cpuinfo_linux_get_processor_package_id(uint32_t processor, uint32_t package
 	}
 }
 
+uint32_t cpuinfo_linux_get_processor_cache_size(uint32_t processor, uint32_t cache_level) {
+        char cache_size_filename[CACHE_FILENAME_SIZE];
+        const int chars_formatted = snprintf(
+                cache_size_filename, CACHE_FILENAME_SIZE, CACHE_FILENAME_FORMAT, processor, cache_level);
+        if ((unsigned int) chars_formatted >= CACHE_FILENAME_SIZE) {
+                cpuinfo_log_warning("failed to format filename for cache size of processor %"PRIu32, processor);
+                return 0;
+        }
+
+        uint32_t cache_size;
+        if (cpuinfo_linux_parse_small_file(cache_size_filename, CACHE_FILESIZE, uint32_parser, &cache_size)) {
+                cpuinfo_log_debug("parsed cache size value of %"PRIu32" KB for level %"PRIu32" for logical processor %"PRIu32" from %s",cache_size, cache_level, processor, cache_size_filename);
+                return cache_size;
+        } else {
+                cpuinfo_log_info("failed to parse cache size for level %"PRIu32"  for processor %"PRIu32" from %s",
```

Comment:
Shouldn't this be an error or at least a warning?
```suggestion
                cpuinfo_log_error("failed to parse cache size for level %"PRIu32"  for processor %"PRIu32" from %s",
```

Reply:
Since I'm anyway falling back to the conservative configuration if the device doesn't publish it, I left it as just info log. And I don't think this is an error, because kernel doesn't mandate it, and there are devices with cache info not populated in acpi or dts tables in kernel.